#!/usr/bin/env python
PACKAGE = "olelidar"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# device
gen.add("scann_ip", str_t, 0, "IP addr to receive packet from.", "192.168.1.100")
gen.add("port", int_t, 1, "UDP port to receive packet from.", 2368)
gen.add("local_ip", str_t, 2, "local pc.", "192.168.1.100")
packet_type_enum = gen.enum([ gen.const("packet_type_A", str_t, "A", "packet type A"),
                              gen.const("packet_type_B", str_t, "B", "packet type B"),
                              gen.const("packet_type_C", str_t, "C", "packettype C")],
                              "An enum to set packet type")
gen.add("packet_type",str_t,3,"Data type of ole lidar","A", edit_method=packet_type_enum)
# gen.add(name, type, level, description, default, min, max)
gen.add("frame_id", str_t, 4, "olelidar")
gen.add("angle_min", double_t, 5, "angle_min",-45,-45,-45)
gen.add("angle_max", double_t, 6, "angle_max",225,225,225)
gen.add("range_min", double_t, 7, "min range", 0.05, 0.05, 0.05)
gen.add("range_max", double_t, 8, "max range", 30.0, 30.0, 30.0)
exit(gen.generate(PACKAGE, PACKAGE, "oleiPuck"))
